audformat/core/table.py

```diff
@@ -316,6 +316,7 @@ def drop_index(
 
         _assert_table_index(table, index, 'drop rows from')
 
+        index = utils.intersect([table.index, index])
```

Comment:
It's a little bit unfortunate that we need to add another step.

I checked if there is a faster implementation by testing:

```python
        new_index = [idx for idx in list(table.index) if idx not in index]
        new_index = utils._alike_index(table.index, new_index)
        table._df = table.df.reindex(new_index)
```
but it took more or less the same time, so I would propose to stay with your implementation.